### PR TITLE
Fix alpha animals projectile message

### PR DESCRIPTION
**Pull request by perkinslr**
## Changes
Use a `Gun_Autopistol` def when a `ProjectileCE` needs to log an impact and doesn't have equipment.

## References
Works around #161 

## Reasoning

Alpha Animals adds the ability for some animals to launch projectiles without equipment.  This means that `equipmentDef` is null.  Alpha Animals works around this by faking a Gun_Autopistol, so we do the same.

## Alternatives

Find where the null `equipmentDef` gets referenced and fix it there.  That would be significantly more work, as it uses a DSL for generating the messages.

## Testing

Check tests you have performed:
- [X] Compiles without warnings
- [X] Game runs without errors
- [X] (For compatibility patches) ...with and without patched mod loaded
- [X] Playtested a colony (minutes)

### CODE REVIEW

**Review by N7Huntsman — DISMISSED** [review]
Fixes the issue in functional testing. I'll approve, but would like a @CombatExtended-Continued/coders to look it over before merging.

File: Source/CombatExtended/CombatExtended/Projectiles/BulletCE.cs [hunk 1]
```diff
@@ -14,16 +14,18 @@ public class BulletCE : ProjectileCE
     {
         private void LogImpact(Thing hitThing, out LogEntry_DamageResult logEntry)
         {
+	    var ed = equipmentDef;
+	    if (ed==null) ed = ThingDef.Named("Gun_Autopistol");
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/Projectiles/BulletCE.cs:18]: Can be simplified to (and then inlined)
```suggestion
	    var ed = equipmentDef ?? ThingDef.Named("Gun_Autopistol");
```

**Review by MadaraUchiha — APPROVED** [review]
LGTM, though consider simplifying the code

**Comment by perkinslr** [comment]
> LGTM, though consider simplifying the code

Good catch!

**Review by N7Huntsman — DISMISSED** [review]

**Review by N7Huntsman — APPROVED** [review]

**Comment by ogromanas** [comment]
XML error: Duplicate XML node name comps in this XML block: <ThingDef ParentName="AnimalThingBase"><defName>AA_Gallatross</defName><label>gallatross</label><description>Throughout every world,
      Flow the tale of old,
      A creature with untold might
      And an immeasurable height.
      Strength that machines can’t copy
      Power that so-called gods call heresy.
      Beware, beware would-be foes
      Beware, the wrath of the Gallatross.</description><statBases><MoveSpeed>5.7</MoveSpeed><ComfyTemperatureMin>-100</ComfyTemperatureMin><MarketValue>4000</MarketValue><PsychicSensitivity>0</PsychicSensitivity><AimingAccuracy>0.6</AimingAccuracy><ArmorRating_Blunt>26</ArmorRating_Blunt><ArmorRating_Sharp>13</ArmorRating_Sharp><MeleeDodgeChance>0.02</MeleeDodgeChance><MeleeCritChance>1.75</MeleeCritChance><MeleeParryChance>0.25</MeleeParryChance></statBases><tools><li Class="CombatExtended.ToolCE"><label>horn</label><capacities><li>Blunt</li></capacities><power>60</power><cooldownTime>2.5</cooldownTime><linkedBodyPartsGroup>HornAttackTool</linkedBodyPartsGroup><armorPenetrationBlunt>40.226</armorPenetrationBlunt></li><li Class="CombatExtended.ToolCE"><capacities><li>Bite</li></capacities><power>48</power><cooldownTime>2.0</cooldownTime><linkedBodyPartsGroup>Teeth</linkedBodyPartsGroup><armorPenetrationSharp>16</armorPenetrationSharp><armorPenetrationBlunt>28</armorPenetrationBlunt></li><li Class="CombatExtended.ToolCE"><label>head</label><capacities><li>Blunt</li></capacities><power>40</power><cooldownTime>2</cooldownTime><linkedBodyPartsGroup>HeadAttackTool</linkedBodyPartsGroup><armorPenetrationBlunt>25</armorPenetrationBlunt><chanceFactor>0.1</chanceFactor></li></tools><verbs><li Class="CombatExtended.VerbPropertiesCE"><verbClass>CombatExtended.Verb_ShootCE</verbClass><hasStandardCommand>true</hasStandardCommand><defaultProjectile>AA_Bullet_Rock</defaultProjectile><warmupTime>2.5</warmupTime><burstShotCount>1</burstShotCount><ticksBetweenBurstShots>2</ticksBetweenBurstShots><minRange>1</minRange><range>30</range><soundCast>AA_RockThrow</soundCast><muzzleFlashScale>0</muzzleFlashScale><ai_IsBuildingDestroyer>true</ai_IsBuildingDestroyer><commonality>1</commonality></li></verbs><race><herdAnimal>false</herdAnimal><predator>false</predator><body>QuadrupedAnimalWithHoovesAndHorn</body><baseBodySize>6.00</baseBodySize><baseHungerRate>5.0</baseHungerRate><baseHealthScale>15</baseHealthScale><foodType>VegetarianRoughAnimal, DendrovoreAnimal</foodType><leatherDef>Leather_Gallatross</leatherDef><willNeverEat><li MayRequire="Ludeon.RimWorld.Royalty">Plant_TreeAnima</li></willNeverEat><wildness>0.99</wildness><manhunterOnDamageChance>1.0</manhunterOnDamageChance><manhunterOnTameFailChance>0.3</manhunterOnTameFailChance><trainability>Advanced</trainability><gestationPeriodDays>75</gestationPeriodDays><litterSizeCurve><points><li>(0.99, 0)</li><li>(1, 1)</li><li>(1.01, 0)</li></points></litterSizeCurve><lifeExpectancy>300</lifeExpectancy><lifeStageAges><li><def>AnimalBaby</def><minAge>0</minAge></li><li><def>AnimalJuvenile</def><minAge>2.0</minAge></li><li><def>AnimalAdult</def><minAge>4</minAge><soundCall>Pawn_Thrumbo_Call</soundCall><soundAngry>Pawn_Thrumbo_Angry</soundAngry><soundWounded>Pawn_Thrumbo_Pain</soundWounded><soundDeath>Pawn_Thrumbo_Death</soundDeath></li></lifeStageAges><soundMeleeHitPawn>Pawn_Melee_BigBash_HitPawn</soundMeleeHitPawn><soundMeleeHitBuilding>Pawn_Melee_BigBash_HitBuilding</soundMeleeHitBuilding><soundMeleeMiss>Pawn_Melee_BigBash_Miss</soundMeleeMiss></race><tradeTags><li>AnimalExotic</li><li>AnimalAlpha</li></tradeTags><modExtensions><li Class="AnimalBehaviours.AnimalStatExtension"><statToAdd><li>AA_PrimaryRole</li><li>AA_RangedAttacks</li></statToAdd><statValues><li>AA_Artillery</li><li>AA_Yes</li></statValues><statDescriptions><li>AA_ArtilleryDesc</li><li>AA_Ranged</li></statDescriptions></li><li Class="CombatExtended.RacePropertiesExtensionCE"><bodyShape>Quadruped</bodyShape></li></modExtensions><comps><li Class="AlphaBehavioursAndEvents.CompProperties_InitialHediff"><hediffname>AA_LowBleed</hediffname><hediffseverity>1</hediffseverity></li></comps><comps><li Class="GiddyUpCore.CompProperties_Overlay"><overlaySide><graphicDataDefault><texPath>Things/Pawn/Animal/Saddles/AA_Saddle5Gallatross_east</texPath><graphicClass>Graphic_Single</graphicClass><drawSize>4.5</drawSize><drawRotated>false</drawRotated></graphicDataDefault><offsetDefault>(0,0,0,0)</offsetDefault></overlaySide><overlayFront><graphicDataDefault><texPath>Things/Pawn/Animal/Saddles/AA_Saddle5Gallatross_south</texPath><graphicClass>Graphic_Single</graphicClass><drawSize>4.5</drawSize><drawRotated>false</drawRotated></graphicDataDefault><offsetDefault>(0,0,0,0)</offsetDefault></overlayFront><overlayBack><graphicDataDefault><texPath>Things/Pawn/Animal/Saddles/AA_Saddle5Gallatross_north</texPath><graphicClass>Graphic_Single</graphicClass><drawSize>4.5</drawSize><drawRotated>false</drawRotated></graphicDataDefault><offsetDefault>(0,0,0,0)</offsetDefault></overlayBack></li></comps></ThingDef>
Verse.Log:Error(String, Boolean)
Verse.XmlInheritance:CheckForDuplicateNodes(XmlNode, XmlNode)
Verse.XmlInheritance:ResolveXmlNodeFor(XmlInheritanceNode)
Verse.XmlInheritance:ResolveXmlNodesRecursively(XmlInheritanceNode)
Verse.XmlInheritance:ResolveXmlNodesRecursively(XmlInheritanceNode)
Verse.XmlInheritance:ResolveXmlNodesRecursively(XmlInheritanceNode)
Verse.XmlInheritance:ResolveXmlNodes()
Verse.XmlInheritance:Resolve()
Verse.LoadedModManager:Verse.LoadedModManager.ParseAndProcessXML_Patch2(XmlDocument, Dictionary`2)
Verse.LoadedModManager:LoadAllActiveMods()
Verse.PlayDataLoader:DoPlayLoad()
Verse.PlayDataLoader:LoadAllPlayData(Boolean)
Verse.<>c:<Start>b__6_1()
Verse.LongEventHandler:RunEventFromAnotherThread(Action)
Verse.<>c:<UpdateCurrentAsynchronousEvent>b__27_0()
System.Threading.ThreadHelper:ThreadStart_Context(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
System.Threading.ThreadHelper:ThreadStart()

**Pull request merged**